### PR TITLE
docs(v2): blog initial draft

### PR DESCRIPTION
**Pull request by dt97**
## Motivation

Addressing #1640, add docs for contributing for v2.

Copied over guides-blog.md from root and reorganized for v2.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/Docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos!)

NA

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/facebook/docusaurus, and link to your PR here.)

#1640

### CODE REVIEW

**Comment by dt97** [comment]
Sure. Would do as mentioned.

**Comment by dt97** [comment]
Thanks for the insights!!!

**Comment by dt97** [comment]
Have updated the blog.md file for v2 as per the insights shared. Kindly re-review the changes.

File: website/docs/blog.md [hunk 1]
```diff
@@ -3,7 +3,112 @@ id: blog
...
+links: [
+    ...
+    {to: 'blog', label: 'Blog', position: 'left'}, // position: 'left' for creating Blog link on the left side of navbar, position: 'right' for creating Blog link on the right side of navbar
+    ...
+]
+```
+
+## Adding Posts
+
+To publish in the blog, create a file within the blog directory with a formatted name of `YYYY-MM-DD-my-blog-post-title.md`. The post date is extracted from the file name.
```
Comment (wgao19) [inline website/docs/blog.md]: shall we mention that the blog directory is by default `blog/` under the project root, and add an "options" section at the end of this doc that goes through the customizable options?

File: website/docs/blog.md [hunk 2]
```diff
@@ -3,7 +3,112 @@ id: blog
...
+- `author` - The text label of the author byline.
+- `authorURL` - The URL associated with the author. This could be a Twitter, GitHub, Facebook account, etc.
+- `authorFBID` - The Facebook profile ID that is used to fetch the profile picture.
+- `authorImageURL` - The URL to the author's image. (Note: If you use both `authorFBID` and `authorImageURL`, `authorFBID` will take precedence. Don't include `authorFBID` if you want `authorImageURL` to appear.)
+- `title` - The blog post title.
+- `unlisted` - The post will be accessible by directly visiting the URL but will not show up in the sidebar in the final build; during local development, the post will still be listed. Useful in situations where you want to share a WIP post with others for feedback.
+
+## Summary Truncation
+
+Use the `<!--truncate-->` marker in your blog post to represent what will be shown as the summary when viewing all published blog posts. Anything above `<!--truncate-->` will be part of the summary. For example:
```
Comment (wgao19) [inline website/docs/blog.md]: the truncate marker is also customizable, [here](https://github.com/facebook/Docusaurus/blob/master/packages/docusaurus-plugin-content-blog/src/index.ts#L42) is the code regarding customizable options

File: website/docs/blog.md [hunk 3]
```diff
@@ -3,7 +3,112 @@ id: blog
...
+But anything from here on down will not be.
+
+Not this.
+
+Or this.
+```
+
+## Advanced Topics
+
+### I want to run in "Blog Only" mode.
```
Comment (wgao19) [inline website/docs/blog.md]: this feels slightly hacky and I'm not sure if the land and redirect behavior is the desired user interaction we want to recommend. @endiliey @yangshun what do you think?

File: website/docs/blog.md [hunk 4]
```diff
@@ -3,7 +3,112 @@ id: blog
...
+
+To publish in the blog, create a file within the blog directory with a formatted name of `YYYY-MM-DD-my-blog-post-title.md`. The post date is extracted from the file name.
+
+For example, at `my-website/blog/2019-09-05-hello-docusaurus-v2.md`:
+
+```yml
+---
+id: hello-docusaurus-v2
+title: Welcome Docusaurus v2
+author: Dattatreya Tripathy
```
Comment (wgao19) [inline website/docs/blog.md]: I would love to credit your contributions on our release notes, but do you mind using a more neutral reference (i.e., you may use Docusaurus directly, the twitter handle is @docusaurus) here for the consistency of Docusaurus?

File: website/docs/blog.md (hunk 3 above)
Comment (dt97) [reply website/docs/blog.md]: Yup here we are creating a static html file which is redirecting to the Blog page whenever the user runs Docusaurus 2 site.

Comment (dt97) [reply website/docs/blog.md]: So what do you suggest. Do I remove this section from the blog.md file?

File: website/docs/blog.md (hunk 4 above)
Comment (dt97) [reply website/docs/blog.md]: Would make the required changes!!!

File: website/docs/blog.md (hunk 2 above)
Comment (dt97) [reply website/docs/blog.md]: Would make the required changes.

File: website/docs/blog.md (hunk 1 above)
Comment (dt97) [reply website/docs/blog.md]: Yes. That would be better I think. Would make the required changes.

File: website/docs/blog.md (hunk 3 above)
Comment (wgao19) [reply website/docs/blog.md]: Discussed with the other maintainers, let's suggest enabling blog only mode with the method used in [Endiliey's blog](https://github.com/endiliey/endiliey.github.io), [here](https://github.com/endiliey/endiliey.github.io/blob/source/docusaurus.config.js#L24) is the relevant code.

Comment (dt97) [reply website/docs/blog.md]: Got it!!! Thanks. Would do.

**Comment by dt97** [comment]
Thanks for the merge. Would try to make the required changes as soon as possible.

**Pull request merged**